### PR TITLE
zoekt-git-index: add cpuprofile and memprofile flags

### PR DESCRIPTION
**Pull request by keegancsmith**
This makes it possible to get profiles for indexing with git. The code is copy-pasta from the pprof godoc. Note: we move main into a function which returns such that the defers run before os.Exit is called.

### CODE REVIEW

**Review by stefanhengl — APPROVED** [review]
Nice!

**Comment by keegancsmith** [comment]
Note: BuildOptions has a MemProfile field which seems to have never been used. This creates a profile per shard. I think I'll actually update this code to use that instead, since this is quite useful to understand memory usage on larger shards / you can make the shard size smaller.

**Comment by keegancsmith** [comment]
@stefanhengl re-requesting review since I made some more fundemental changes.

File: build/builder.go [hunk 1]
```diff
@@ -168,6 +168,7 @@ func (o *Options) Flags(fs *flag.FlagSet) {
 	fs.StringVar(&o.IndexDir, "index", x.IndexDir, "directory for search indices")
 	fs.BoolVar(&o.CTagsMustSucceed, "require_ctags", x.CTagsMustSucceed, "If set, ctags calls must succeed.")
 	fs.Var(largeFilesFlag{o}, "large_file", "A glob pattern where matching files are to be index regardless of their size. You can add multiple patterns by setting this more than once.")
+	fs.StringVar(&o.MemProfile, "memprofile", "", "write memory profile(s) to `file.shardnum`. Note: sets parallelism to 1.")
```
Comment (stefanhengl) [inline build/builder.go:171]: nit: maybe we can find a description that makes it more obvious that the value of the flag is the filename of the mem profile? I don't have a good suggestion, so feel free to ignore this.

**Review by stefanhengl — APPROVED** [review]

File: build/builder.go (hunk 1 above)
Comment (keegancsmith) [reply build/builder.go:171]: yeah I also tried to think up a better description. This is a tiny mutation on the default description suggested by the godoc. I can't really think of something better so just gonna go ahead and keep it.

**Pull request merged**